clients/drcachesim/docs/drcachesim.dox.in

```diff
@@ -1010,48 +1010,109 @@ architecture to achieve optimal efficiency.  Google is sharing
-efficiency of this important class of workloads.
+efficiency of this important class of workloads. To protect Google's
+intellectual property, these traces have been modified in order to
+filter out sensitive information. These traces follow a synthetic ISA
+(#DR_ISA_REGDEPS) that removes architecture specific details (e.g., the
+opcode of instructions), while still providing enough information (e.g.,
+register dependencies) to perform meaningful analyses and simulations.
 
-\section sec_google_format Trace Format
+\section sec_google_format Public Trace Format
 
 The Google workload traces are captured using DynamoRIO's
 [drmemtrace](@ref page_drcachesim).  The traces are records of
 instruction and memory accesses as described at \ref
-sec_drcachesim_format.  We separate instruction and memory access
-records from each software thread into a separate file
-(.memtrace.gz). In addition, for each software thread, we also provide
-a branch_trace which contains execution data (taken/not taken, branch
-target) about each branch instruction (conditional, non-conditional,
-calls, etc.).  Finally, for each workload trace, we provide a thread
-statistics file (.threadstats.csv) which contains the thread ID (tid),
-instruction count, non-fetched instruction count (e.g. implicit
-instructions generated from microcode), load count, store count, and
-prefetch count.
+sec_drcachesim_format. While memory accesses are left unchanged
+compared to the original trace, instructions follow the
+#DR_ISA_REGDEPS synthetic ISA.
+
+#DR_ISA_REGDEPS has the purpose of preserving register dependencies and giving
+hints on the type of operation an instruction performs.
+
+Being a synthetic ISA, some routines that work on instructions coming from an
+actual ISA (such as #DR_ISA_AMD64) are not supported (e.g., decode_sizeof()).
+
+Currently we support:
+- instr_convert_to_isa_regdeps(): to convert an #instr_t of an actual ISA to a
+  #DR_ISA_REGDEPS #instr_t.
+- instr_encode() and instr_encode_to_copy(): to encode a #DR_ISA_REGDEPS #instr_t
+  into a sequence of contiguous bytes.
+- decode() and decode_from_copy(): to decode an encoded #DR_ISA_REGDEPS instruction
+  into an #instr_t.
+
+A #DR_ISA_REGDEPS #instr_t contains the following information:
+- categories: composed by #dr_instr_category_t values, they indicate the type of
+  operation performed (e.g., a load, a store, a floating point math operation, a
+  branch, etc.).  Note that categories are composable, hence more than one category
+  can be set.  This information can be obtained using instr_get_category().
+- arithmetic flags: we don't distinguish between different flags, we only report if
+  at least one arithmetic flag was read (all arithmetic flags will be set to read)
+  and/or written (all arithmetic flags will be set to written).  This information
+  can be obtained using instr_get_arith_flags().
+- number of source and destination operands: we only consider register operands.
```

Comment:
Mention that memory operands can be deduced by subsequent read and write records.